src/azul/service/responseobjects/hca_response_v5.py

```diff
@@ -329,70 +329,34 @@ class ProjectSummaryResponse(BaseSummaryResponse):
     """
 
     @classmethod
-    def get_bucket_terms(cls, project_id, project_buckets, agg_key):
+    def get_distinct_terms(cls, items, property_name):
         """
-        Return a list of the keys of the buckets from an ElasticSearch aggregate
-        of a given project with the format:
-        {
-          "buckets": [
-            {
-              "key": $project_id,
-              $agg_key: {
-                "buckets": [
-                  {
-                    "key": "a",
-                    "doc_count": 2
-                  }
-                ]
-              }
-            },
-            ...
-          ]
-        }
+        Iterate through the list of items (dicts) and return a list of distinct values
+        in all the items with property_name as a key
 
-        :param project_id: string UUID of the project info to retrieve
-        :param project_buckets: A dictionary from an ElasticSearch aggregate
-        :param agg_key: Key of aggregation to use
-        :return: list of bucket keys
-        """
-        for project_bucket in project_buckets['buckets']:
-            if project_bucket['key'] != project_id:
-                continue
-            return [bucket['key'] for bucket in project_bucket[agg_key]['buckets']]
-        return []
-
-    @classmethod
-    def get_bucket_value(cls, project_id, project_buckets, agg_key):
-        """
-        Return a value of the bucket of the given project from an
-        ElasticSearch aggregate with the format:
-        {
-          "buckets": [
-            {
-              "key": $project_id,
-              $agg_key: {
-                "value" : value
-              }
-            },
-            ...
-          ]
-        }
+        The value at the given property name must be a list of hashable items
 
-        :param project_id: string UUID of the project info to retrieve
-        :param project_buckets: A dictionary from an ElasticSearch aggregate
-        :param agg_key: Key of aggregation to use
-        :return: value in given project
+        :param AttrList items: list of dicts to extract distinct values from
+        :param str property_name: key of each item to extract
+        :return: list of distinct values
         """
-        for project_bucket in project_buckets['buckets']:
-            if project_bucket['key'] == project_id:
-                return project_bucket[agg_key]['value']
-        return -1
+        terms = set()
+        for item in items:
+            try:
+                terms |= set(item[property_name])
+            except KeyError:
```

Comment:
I made the assumption that it's acceptable that some items not have the property since this is the case for the real data.  This also means that there is no error if we try to look for a property that does not exist in any of the items.  I don't see an issue doing this but I wanted to confirm that this is acceptable.

Reply:
I think the real data should not have missing properties. If it does, that would constitute a bug. I want to avoid the ambiguity between a property missing from an entity (aggregate entity), and it's value being `None` (`[None]`). We should invariably guarantee that every entity always has all properties so they behave more like instance of a class (with a fixed set of attributes that can be none but are always present) instead of a dictionary where an entry can be absent or None.

Reply:
In this case, suppose that one of the value from `item[property_name]` is `None`. Should `terms` eventually include or exclude `None`?

For example,
* `[None]` → `[]`?
* `[123, None ,456]` → `[123, 456]`?

Reply:
After reconsidering the case, I don't know if it is possible to see `None` in `item[property_name]`. Please disregard my comment.